internal/provider/string.go

```diff
@@ -257,3 +322,67 @@ func generateRandomBytes(charSet *string, length int) ([]byte, error) {
 func readNil(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	return nil
 }
+
+func resourceStateUpgradeAddNumeric(_ context.Context, rawState map[string]interface{}, _ interface{}, resourceName string) func(_ context.Context, rawState map[string]interface{}, _ interface{}) (map[string]interface{}, error) {
+	return func(_ context.Context, rawState map[string]interface{}, _ interface{}) (map[string]interface{}, error) {
+		if rawState == nil {
+			return nil, fmt.Errorf("resource %s state upgrade failed, state is nil", resourceName)
+		}
+
+		number, ok := rawState["number"].(bool)
+		if !ok {
+			return nil, fmt.Errorf("resource %s state upgrade failed, number could not be asserted as bool: %T", resourceName, rawState["number"])
+		}
+
+		rawState["numeric"] = number
+
+		return rawState, nil
+	}
+}
+
+func customDiffIfValue(key string) func(context.Context, *schema.ResourceDiff, interface{}) error {
+	return customdiff.IfValue(
+		key,
+		func(ctx context.Context, value, meta interface{}) bool {
+			return !value.(bool)
+		},
+		func(_ context.Context, d *schema.ResourceDiff, _ interface{}) error {
+			vm := d.GetRawConfig().AsValueMap()
+			if vm["number"].IsNull() && vm["numeric"].IsNull() {
+				err := d.SetNew("number", true)
+				if err != nil {
+					return err
+				}
+				err = d.SetNew("numeric", true)
+				if err != nil {
+					return err
+				}
+			}
+			return nil
+		},
+	)
+}
+
+func customDiffIfValueChangeNumber() func(context.Context, *schema.ResourceDiff, interface{}) error {
```

Comment:
Same as per function above (and below).

Just something to explain the whole "if `number` has changed, we want to prograpage to `numeric`" and vice-versa.

Reply:
Agreed. Added.